File: HomeEnergyUsageTracker/src/HomeEnergyUsageTracker.Core/Model/UsageAggregate/Usage.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.

namespace HomeEnergyUsageTracker.Core;

public class Usage
{
    public Guid UsageId { get; set; }

    /// <summary>
    /// Gets or sets the tenant ID for multi-tenant isolation.
    /// </summary>
    public Guid TenantId { get; set; }
    public Guid UtilityBillId { get; set; }
    public DateTime Date { get; set; }
    public decimal Amount { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: HomeEnergyUsageTracker/src/HomeEnergyUsageTracker.WebApp/projects/home-energy-usage-tracker/src/app/models/usage.ts
export interface Usage {
  usageId: string;
  utilityBillId: string;
  date: Date;
  amount: number;
  createdAt: Date;
}

export interface CreateUsageRequest {
  utilityBillId: string;
  date: Date;
  amount: number;
}

export interface UpdateUsageRequest {
  usageId: string;
  utilityBillId: string;
  date: Date;
  amount: number;
}
